modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentDataProcessor.java

```diff
@@ -0,0 +1,47 @@
+package com.enonic.xp.core.impl.content;
+
+import java.util.EnumSet;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.enonic.xp.content.ContentInheritType;
+import com.enonic.xp.content.ContentPropertyNames;
+import com.enonic.xp.data.PropertyTree;
+import com.enonic.xp.node.NodeDataProcessor;
+
+public abstract class ContentDataProcessor
+    implements NodeDataProcessor
+{
+    public PropertyTree process( final PropertyTree originalData )
+    {
+        final PropertyTree data = originalData.copy();
+
+        if ( data.hasProperty( ContentPropertyNames.INHERIT ) )
+        {
+            boolean changed = false;
+            final ImmutableSet.Builder<String> inheritSet = ImmutableSet.builder();
+
+            for ( final String value : data.getStrings( ContentPropertyNames.INHERIT ) )
+            {
+                if ( !getTypesToProceed().contains( ContentInheritType.valueOf( value ) ) )
+                {
+                    inheritSet.add( value );
+                }
+                else
+                {
+                    changed = true;
+                }
+            }
+
+            if ( changed )
+            {
+                data.removeProperties( ContentPropertyNames.INHERIT );
+                data.addStrings( ContentPropertyNames.INHERIT, inheritSet.build() );
+            }
+        }
+
+        return data;
+    }
+
+    abstract protected EnumSet<ContentInheritType> getTypesToProceed();
```

Comment:
Codacy found an issue: ['protected' modifier out of order with the JLS suggestions.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6080104)